homeassistant/components/fritz/switch.py

```diff
@@ -0,0 +1,486 @@
+                NewEnable="1" if turn_on else "0",
+            )
+        )
+
+
+class FritzBoxProfileSwitch(FritzBoxBaseSwitch, SwitchEntity):
+    """Defines a FRITZ!Box Tools DeviceProfile switch."""
+
+    def __init__(self, fritzbox_tools, device_friendly_name, profile):
+        """Init Fritz profile."""
+        self.fritzbox_tools: FritzBoxTools = fritzbox_tools
+        self.profile = profile
+
+        switch_info = SwitchInfo(
+            description=f"Profile {profile}",
+            friendly_name=device_friendly_name,
+            icon="mdi:router-wireless-settings",
+            type=SWITCH_TYPE_DEVICEPROFILE,
+            callback_update=self._async_fetch_update,
+            callback_switch=self._async_switch_on_off_executor,
+        )
+        super().__init__(self.fritzbox_tools, switch_info)
+
+    async def _async_fetch_update(self):
+        """Update data."""
+        try:
+            status = await self.hass.async_add_executor_job(
+                self.fritzbox_tools.fritz_profiles[self.profile].get_state
+            )
+            _LOGGER.debug(
+                "Specific %s response: get_State()=%s",
+                SWITCH_TYPE_DEVICEPROFILE,
+                status,
+            )
+            if status == "never":
+                self._attr_is_on = False
+                self._is_available = True
+            elif status == "unlimited":
+                self._attr_is_on = True
+                self._is_available = True
+            else:
+                self._is_available = False
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.error("Could not get %s state", self.name, exc_info=True)
+            self._is_available = False
+
+    async def _async_switch_on_off_executor(self, turn_on: bool) -> None:
+        """Handle profile switch."""
+        state = "unlimited" if turn_on else "never"
+        await self.hass.async_add_executor_job(
+            lambda: self.fritzbox_tools.fritz_profiles[self.profile].set_state(state)
+        )
+
+
+class FritzBoxWifiSwitch(FritzBoxBaseSwitch, SwitchEntity):
+    """Defines a FRITZ!Box Tools Wifi switch."""
+
+    def __init__(self, fritzbox_tools, device_friendly_name, network_num, network_name):
+        """Init Fritz Wifi switch."""
+        self._fritzbox_tools: FritzBoxTools = fritzbox_tools
+
+        self._attributes = {}
+        self._network_num = network_num
+
+        switch_info = SwitchInfo(
+            description=f"Wi-Fi {network_name}",
+            friendly_name=device_friendly_name,
+            icon="mdi:wifi",
+            type=SWITCH_TYPE_WIFINETWORK,
+            callback_update=self._async_fetch_update,
+            callback_switch=self._async_switch_on_off_executor,
+        )
+        super().__init__(self._fritzbox_tools, switch_info)
+
+    async def _async_fetch_update(self):
+        """Fetch updates."""
+
+        try:
+            wifi_info = await self.hass.async_add_executor_job(
```

Comment:
Only put the code in the try block that can throw an exception

Reply:
Done with latest cleanup